File: Hash Project-IvanFonseca/project4.scala
import scala.io.Source
import scala.collection.mutable.ArrayBuffer
import java.io.{ FileNotFoundException, IOException}
import java.lang.{ ArrayIndexOutOfBoundsException }
import java.io.PrintWriter
import scala.io.StdIn.{readLine, readInt}
object project4 {
  val arrayBuffer1 = ArrayBuffer[String]()
  val arrayBuffer = ArrayBuffer[Word]()
  //Word object were has the word and times it occurs
  class Word(var word : String,var times : Int){
    this.setWord(word)
    this.setTimes(times)
   
    def getWord() : String = word
    def setWord(word : String) = this.word=word
    def getTimes() : Int = times
    def setTimes(times : Int) = this.times=times
    
    override def toString() : String={
      return "Word: %s occurs %d times".format(this.word,this.times) 
    }
  }
  
   def main(args : Array[String] ){
       //Creates input.txt file with 
       //content of the assignment description    
       var fileName = "input.txt"
       var fileContent ="For this assignment you are to use read in this text for your input (input.txt) for input to your scala and java program. In this project you are to read each word in from the assignment file and output the list of unique words and the number of times each word comes up within the assignment file. For both programs you are to create two types of classes: one class that manages all the word objects, and another class whose instance takes in a unique word through its constructor function. This word encapsulating class must contain a member function update that increments its word count every time this word is found within the text. Points will be given based on producing the correct output, creating the manager and container class and lastly for style. This assignment is worth 20 pts."
       //Passes to the fileIO function were
       //it reads the input.txt file and puts
       //every word into an array
       fileIO(fileName,fileContent)
       //ArrayBuffer for unique words
       val uniqueBuffer = ArrayBuffer[Word]() 
       var count = 0
       //double for loop that sees how many times
       // the word occurs in the array
       for(element <- arrayBuffer1){
           count=0
           for(e <- arrayBuffer1){
              //if the word occurs then the count 
              // is incremented
              if(element==e){
                count=count+1
              }
         }
           //If occured more than once than added to
           // word list
           if(count>1){
             var word = new Word(element,count)
             arrayBuffer += word
           }
           //If occured once than added to
           // unique list
           else{
             var word = new Word(element,count)
             uniqueBuffer += word
           } 
       }
       //Prints how many times a word occurs
       println("List of Words:")
        for(element <- arrayBuffer){
          println(element)
        }
        //Prints unique words
        println("List of Unique Words:")
        for(element <- uniqueBuffer){
          println(element+" (unique)")
        }
      }
     def fileIO(fileName : String, fileContent : String){
       val writer = new PrintWriter(fileName)
       writer.write(fileContent)
       writer.close()
       val file =  Source.fromFile(fileName)
       val lineIterator = file.getLines
       //Adds each word into an array buffer
       for(line <- lineIterator)
         for(element <- line.split(" ") ){
             arrayBuffer1 += element
     }
       file.close()
     }
}

File: Hash Project-IvanFonseca/project4.java
import java.util.*;
import java.io.*;
 
class HashObject {
    public String element;
    public int count = 1;
    // Constructor
    HashObject(String input) {
            element = input;   
    }

    // Find appropriate base class to override that include the hashCode
    // and equals member functions
    public int hashCode() {
            //System.out.println(" Code " + element.hashCode() + " Element " + element );
            return element.hashCode();
    }       
    public boolean equals(Object inputParam) {
       HashObject inputObject = (HashObject) inputParam;
       //System.out.println(element + " - = - " + inputObject.element);
       if (element.equals( inputObject.element)) {	
    	   //System.out.println("The current element is " + hashObject3element      
    	   inputObject.count++;
           return true;
        }
            else {
            	return false;
            }
    }
}
 
class project4{
	
	
	static List<String> list = new ArrayList<String>();
    public static void main(String args[]) {
    	project4 p4 =new project4();
    	HashSet<HashObject> hashSet = new HashSet<HashObject>();
       
        p4.printFile("input.txt");
        //iterates through Array list of words and
        //adds them to hash set.
        for(String s:list) {
        	
        	HashObject hashObject = new HashObject(s);
            hashSet.add(hashObject);  
        }
        p4.hashSetPrint(hashSet);
    }
    
    public void hashSetPrint(HashSet<HashObject> hashSet){
    	Iterator<HashObject> iter = hashSet.iterator();
        HashObject hashObject3;
        //Prints how many times a word occurs.
        while (iter.hasNext()) {
                hashObject3 = iter.next();
                if(hashObject3.count !=1){
                	System.out.println("Word: " + hashObject3.element + " Occurs " + hashObject3.count +" Times");
                }
        } 
        Iterator<HashObject> iter2 = hashSet.iterator();
        HashObject hashObject4;
        //Prints Unique Words.
        while (iter2.hasNext()) {
            hashObject4 = iter2.next();
            if(hashObject4.count ==1){
            	System.out.println( hashObject4.element +" is a Unique Word");
            }
        } 
    }
    public void printFile(String fileName) {
    	try {
            File file = new File(fileName);
            FileReader fr = new FileReader(file);
            Scanner inputStream = new Scanner(fr);         
            while (inputStream.hasNextLine()) {        
                String line = inputStream.next();
          
             for ( String element : line.split(" ")) {
            	 //Adds each word into an array
            	 list.add(element);
                }
            }
            fr.close();
            inputStream.close();
        } catch ( IOException e ) {
            System.out.println(" Exception error -- " );
        } catch ( java.util.NoSuchElementException e ) {
            System.out.println(" At the end of the file " );
        }
    }
}